lib/merit/flex/with_satisfied_demand.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Merit
+  module Flex
+    # Flex participant with demand partially satisfied by a given satisfied_demand_curve.
+    # The direct satisfied demand is not price setting.
+    #
+    # A curve with negative values represents satisfied input.
```

Comment:
Not sure how price-setting works, but if there is e.g. 2 MWh in a specific hour of otherwise curtailed wind output and 2 MWh of electricity input to the electrolyser, both these 2 MWh are irrelevant for price-setting (they can be crossed out). It seems from this comment that only the flex **demand** participant is excluded from price-setting here.

Reply:
Update: when testing this, it seems to work as expected.